### DISCUSSION TITLE
FastAPI for Julia

### DISCUSSION THREAD

**Discussion by rozeappletree** [fastapi/fastapi#7294] [Questions] [answered]
Hi,

For Julia, I am planning to implement a wrapper around HTTP.jl that works like FastAPI. Can I know where can I find **architecture and design(mainly)** of FastAPI and the key algorithmic/processing factors that make it efficient.

Thanks,
Rakesh

**Comment by raphaelauv** [comment, marked as answer]
Basically fastAPI is a wrapper of starlette doing endpoint (input and ouput ) data validation based on typing thank to pydantic

this show a basic architecture of fastAPI -> https://github.com/tiangolo/fastapi/issues/935#issuecomment-691478830

Also look to the documention of respective projects :

https://fastapi.tiangolo.com/
https://www.starlette.io/
https://pydantic-docs.helpmanual.io/

The name of your issue is not clear , maybe rename it -> "How build a fastAPI equivalent in Julia"

**Comment by tiangolo** [comment]
Thanks for the help here @raphaelauv ! :clap: :bow: 

Thanks for reporting back and closing the issue @rakesh4real  :+1:
